src/status_im2/contexts/onboarding/create_password/view.cljs

```diff
@@ -1,42 +1,197 @@
-            [utils.re-frame :as rf]))
-
-(defn navigation-bar
+  (:require
+    [quo2.core :as quo]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im2.common.resources :as resources]
+    [status-im2.contexts.onboarding.common.navigation-bar.view :refer [navigation-bar]]
+    [status-im2.contexts.onboarding.create-password.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn background-image
   []
-  [rn/view {:style style/navigation-bar}
-   [quo/page-nav
-    {:align-mid?            true
-     :mid-section           {:type :text-only :main-text ""}
-     :left-section          {:type                :blur-bg
-                             :icon                :i/arrow-left
-                             :icon-override-theme :dark
-                             :on-press            #(rf/dispatch [:navigate-back])}
-     :right-section-buttons [{:type                :blur-bg
-                              :icon                :i/info
-                              :icon-override-theme :dark
-                              :on-press            #(js/alert "Pending")}]}]])
-
-(defn page
+  [rn/image
+   {:style       style/image-background
+    :blur-radius 13
+    :source      (resources/get-image :onboarding-blur-bg)}])
+
+(defn header
   []
-  [rn/view {:style style/page-container}
-   [navigation-bar]
-   [rn/view {:style {:padding-horizontal 20}}
-    [quo/text
-     {:size   :heading-1
-      :weight :semi-bold
-      :style  {:color colors/white}} "Create profile password"]
-    [quo/button
-     {:on-press #(rf/dispatch [:navigate-to :enable-biometrics])
-      :style    {}} (i18n/label :t/continue)]]])
+  [rn/view {:style style/heading}
+   [quo/text
+    {:style  style/heading-title
+     :weight :semi-bold
+     :size   :heading-1}
+    (i18n/label :t/password-creation-title)]
+   [quo/text
+    {:style  style/heading-subtitle
+     :weight :regular
+     :size   :paragraph-1}
+    (i18n/label :t/password-creation-subtitle)]])
+
+(defn password-with-hint
+  [{{:keys [text status shown]} :hint :as input-props}]
+  [rn/view
+   [quo/input
+    (-> input-props
+        (dissoc :hint)
+        (assoc :type           :password
+               :override-theme :dark
+               :blur?          true))]
+   [rn/view {:style style/label-container}
+    (when shown
+      [:<>
+       [quo/icon (if (= status :success) :i/check-circle :i/info)
+        {:container-style style/label-icon
+         :color           (style/label-icon-color status)
+         :size            16}]
+       [quo/text
+        {:style (style/label-color status)
+         :size  :paragraph-2}
+        text]])]])
+
+(defn password-inputs
+  [{:keys [passwords-match? on-change-password on-change-repeat-password on-input-focus
+           password-long-enough? empty-password? same-password-lenght?]}]
+  (reagent/with-let [show-hint-2? (reagent/atom false)]
```

Comment:
I see it as syntax sugar for creating form-2 components. Deep down the macro does more stuff https://github.com/reagent-project/reagent/blob/a14faba55e373000f8f93edfcfce0d1222f7e71a/src/reagent/ratom.clj#L20

I'm not against it, but it can be a bit confusing because some devs might accidentally add bindings that should be local to the renderer function. Maybe we should avoid this macro? No strong opinion.

Reply:
@flexsurfer 
I used it because it's a less-nested way to create form-2 components.

@ilmotta yes, it provide us a `finally` clause at the end of the body, we can use it to set instructions that will be executed when the component is unmounted.
e.g. remove a listener, delete a timeout, call a re-frame event, etc.

I can remove it if you want and stop using it, whatever you think is better for us. :+1:

Reply:
It's good to have `with-let` in our toolbox, and it's good to see new approaches to old problems.

It does look like we don't need the cleanup feature of the macro, so a bread and butter form-2 component should be enough.

@ulisesmac, if you find other places where the macro can be useful, please share. If I'm not mistaken we do have some places that use `componentWillUnmount`, and these might be good cases for `with-let`. Yet another option is to do cleanup in a callback returned from within `react-native/use-effect` (of course this approach is not always convenient).

Reply:
@ilmotta Got it, I agree.

FYI recently I suggested its use in a @J-Son89 PR, it replaced hook uses, if you want to take a look:
https://github.com/status-im/status-mobile/blob/03cf4cec0e0e8c1026978f64539f7db76319710a/src/status_im2/contexts/onboarding/common/intro/view.cljs#L52

Reply:
I see the `finally` there, cool stuff @ulisesmac!